qbraid/interface/qbraid_qasm/test/test_qasm_utils.py

```diff
@@ -48,6 +48,51 @@ def _check_output(output, expected):
     assert actual_circuit == expected_circuit
 
 
+def test__qasm_random():
+    """test random circuit"""
+
+    # Test case 1: Generate a random circuit with default parameters
+    circuit = _qasm_random()
+    assert qasm_num_qubits(circuit) >= 1
+
+    # Test case 2: Generate a random circuit with specified parameters
+    circuit = _qasm_random(num_qubits=3, depth=3, max_operands=3, seed=42)
+    assert qasm_num_qubits(circuit) == 3
+
+    # Test case 3: Generate a random circuit with measurement
+    circuit = _qasm_random(measure=True)
+    assert qasm_num_qubits(circuit) >= 1
+
+    # Test case 3: Generate a random circuit with measurement
+    circuit = _qasm_random(num_qubits=3, depth=3, max_operands=3, seed=42, measure=True)
+    assert qasm_num_qubits(circuit) == 3
+
+    out__expected = """
+// Random Circuit generated by qBraid
+OPENQASM 3.0;
+include "stdgates.inc";
+/*
+    seed = 42
+    num_qubits = 3
+    depth = 3
+    max_operands = 3
+*/
+qubit[3] q;
+bit[3] c;
+y q[0];
+crx(5.3947298351621535) q[1],q[2];
+cz q[0],q[2];
+t q[1];
+cp(0.8049616944763924) q[1],q[2];
+u1(2.829858307545725) q[0];
+c[0] = measure q[0];
+c[1] = measure q[1];
+c[2] = measure q[2];
+
+"""
+    _check_output(circuit, out__expected)
```

Comment:
I guess I'm a little confused here, because if the function truly generates a random circuit, how can you know the expected output beforehand? Shouldn't it change each time you run it?

Reply:
Oh, I see now. This is the expected output for random seed value `seed=42`. That makes sense.